mmdeploy/backend/coreml/torchscript2coreml.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+
+from typing import Dict, Sequence, Union
+
+import coremltools as ct
+import torch
+
+from mmdeploy.utils import get_root_logger
+
+try:
+    # user might need ops from torchvision
+    import torchvision  # noqa
+except ImportError:
+    pass
+
+
+def get_model_suffix(convert_to: str) -> str:
+    assert convert_to == 'neuralnetwork' or convert_to == 'mlprogram'
+    suffix = ''
+    if convert_to == 'neuralnetwork':
+        suffix = '.mlmodel'
+    if convert_to == 'mlprogram':
+        suffix = '.mlpackage'
+    return suffix
+
+
+def create_shape(name: str, input_shapes: Dict) -> ct.Shape:
+    """Create input shape."""
+    min_shape = input_shapes['min_shape']
+    max_shape = input_shapes['max_shape']
+    default_shape = input_shapes['default_shape']
+    assert len(min_shape) == len(max_shape) == len(default_shape)
+    shape = []
+    n_dim = len(min_shape)
+    for i in range(n_dim):
+        low = min_shape[i]
+        high = max_shape[i]
+        assert low <= high
+        if low == -1 or high == -1:
+            shape.append(ct.RangeDim())
+        elif low == high:
+            shape.append(low)
+        else:
+            shape.append(ct.RangeDim(low, high))
+
+    shape = ct.Shape(shape=shape, default=default_shape)
+    return ct.TensorType(shape=shape, name=name)
+
+
+def from_torchscript(torchscript_model: Union[str,
+                                              torch.jit.RecursiveScriptModule],
+                     output_file_prefix: str,
+                     input_names: Sequence[str],
+                     output_names: Sequence[str],
+                     input_shapes: Dict,
+                     convert_to: str = 'neuralnetwork',
+                     fp16_mode: bool = False,
+                     skip_model_load: bool = True,
+                     **kwargs):
+    """Create a coreml engine from torchscript.
+
+    Args:
+        torchscript_model (Union[str, torch.jit.RecursiveScriptModule]):
+            The torchscript model to be converted.
+        output_file_prefix (str): The output file prefix.
+        input_names (Sequence[str]): The input names of the model.
+        output_names (Sequence[str]): The output names of the model.
+        input_shapes (Dict): The input shapes include max_shape, min_shape and
+            default_shape
+        convert_to (str, optional): The converted model type, can be
+            'neuralnetwork' or 'mlprogram'. Defaults to 'neuralnetwork'.
+        fp16_mode (bool, optional): Convert to fp16 model. Defaults to False.
+        skip_model_load (bool, optional): Skip model load. Defaults to True.
+    """
+
+    try:
+        from mmdeploy.backend.torchscript import get_ops_path
+        torch.ops.load_library(get_ops_path())
+    except Exception:
+        get_root_logger().warning(
```

Comment:
Please print exception e to give more tips.

Reply:
@grimoire